python_example/consumer.py

```diff
@@ -0,0 +1,48 @@
+import time
+import os
+from confluent_kafka import Consumer
+
+import requests
+
+from dotenv import load_dotenv
+
+load_dotenv()
+
+def _get_token(config):
+    payload = {
+        'grant_type': 'client_credentials',
+        'scope': 'api.iam.service_accounts'
+    }
+    resp = requests.post(os.getenv("RHOAS_SERVICE_ACCOUNT_OAUTH_TOKEN_URL"),
+                         auth=(os.getenv("RHOAS_SERVICE_ACCOUNT_CLIENT_ID"), os.getenv("RHOAS_SERVICE_ACCOUNT_CLIENT_SECRET")),
+                         data=payload)
+    token = resp.json()
+    return token['access_token'], time.time() + float(token['expires_in'])
+
+
+topic = "prices"
+consumer_conf = {
+    'bootstrap.servers': os.getenv("KAFKA_HOST"),
+    'security.protocol': 'SASL_SSL',
+    'sasl.mechanisms': 'OAUTHBEARER',
+    "group.id": "test-group",
+    'session.timeout.ms': 6000,
+    'auto.offset.reset': 'earliest',
+    'oauth_cb': _get_token,
+}
+consumer = Consumer(consumer_conf)
+
+consumer.subscribe([topic])
+
+while True:
+    try:
+        # SIGINT can't be handled when polling, limit timeout to 1 second.
```

Comment:
I think you can remove this comment and let ppl kill the process with `Ctrl + C`, you can remove the `except` and run `consumer.close` in a `finally`

Reply:
Not able to close it with a `finally`

Reply:
Can you share the problem?

Reply:
Code snippet:
```
while True:
    try:
        msg = consumer.poll(1.0)
        if msg is not None:
            print(msg.value())
    finally:
        consumer.close()
```
Error:
```
Traceback (most recent call last):
  File "python_example/consumer.py", line 48, in <module>
    msg = consumer.poll(1.0)
RuntimeError: Consumer closed
```